File: src/main/java/me/tomasan7/jecnadesktop/parser/parsers/GradesParser.java
package me.tomasan7.jecnadesktop.parser.parsers;

import me.tomasan7.jecnadesktop.data.Grades;

/**
 * Is responsible for parsing any kind of formatted {@link String} (eg. JSON, XML) to {@link Grades} instance.
 */
public interface GradesParser
{
	Grades parse (String source);
}

File: app/src/main/kotlin/me/tomasan7/jecnadesktop/parser/parsers/GradesParser.kt
package me.tomasan7.jecnadesktop.parser.parsers

import me.tomasan7.jecnadesktop.data.Grades

/**
 * Is responsible for parsing any kind of formatted [String] (eg. JSON, XML) to [Grades] instance.
 */
interface GradesParser
{
    /**
     * @throws me.tomasan7.jecnadesktop.parser.ParseException When the source isn't in correct format.
     */
    fun parse(source: String): Grades
}